meson.build

```diff
@@ -9,25 +9,22 @@ project(
     ],
 )
 
-add_global_arguments(
-    [
-        '-DGETTEXT_PACKAGE="@0@"'.format(meson.project_name()),
-    ],
-    language: 'c',
-)
-
 config = configuration_data()
-config.set('EXEC_NAME', meson.project_name())
-config.set('GETTEXT_PACKAGE', meson.project_name())
-config.set('RESOURCES', '/' + '/'.join(meson.project_name().split('.')) + '/')
-config.set('VERSION', meson.project_version())
-config.set('PREFIX', get_option('prefix'))
-config.set('NAME', 'Tooth')
-config.set('WEBSITE', 'https://github.com/GeopJr/tooth')
-config.set('SUPPORT_WEBSITE', 'https://github.com/GeopJr/tooth/issues')
-config.set(
-    'COPYRIGHT',
-    '© 2018-2021 bleak_grey\n© 2022 Evangelos \"GeopJr\" Paterakis',
+config.set_quoted('EXEC_NAME', meson.project_name())
+config.set_quoted('GETTEXT_PACKAGE', meson.project_name())
+config.set_quoted('DOMAIN', meson.project_name ())
+config.set_quoted('RESOURCES', '/' + '/'.join(meson.project_name().split('.')) + '/')
+config.set_quoted('VERSION', meson.project_version())
+config.set_quoted('PREFIX', get_option('prefix'))
+config.set_quoted('NAME', 'Tooth')
+config.set_quoted('WEBSITE', 'https://github.com/GeopJr/tooth')
+config.set_quoted('SUPPORT_WEBSITE', 'https://github.com/GeopJr/tooth/issues')
+configure_file(output: 'config.h', configuration: config)
+
+add_project_arguments (
+  '--include','config.h',
+  '-DGETTEXT_PACKAGE="@0@"'.format(meson.project_name()),
+  language: 'c'
```

Comment:
I know `c` was already used before, but I'm wondering whether it shouldn't be `vala` instead 
as it is mentioned as a supported language string in the [docs](https://mesonbuild.com/Reference-manual_functions.html#project)

Reply:
Vala is supported, but those arguments are for the C compiler, as it asking to include the C header that was generated using `configure_file`. On the other hand, we will use that C Header in Vala through the VAPI that is included in the project files